tests/test_transport_proxy.py

```diff
@@ -178,6 +178,46 @@ def test_send_grpc_with_meta():
     ray.shutdown()
 
 
+def test_send_grpc_with_party_specific_meta():
+    compatible_utils.init_ray(address='local')
+    cluster_config = {
+        constants.KEY_OF_CLUSTER_ADDRESSES: "",
+        constants.KEY_OF_CURRENT_PARTY_NAME: "",
+        constants.KEY_OF_TLS_CONFIG: "",
+        constants.KEY_OF_CROSS_SILO_MESSAGES_MAX_SIZE_IN_BYTES: None,
+        constants.KEY_OF_CROSS_SILO_SERIALIZING_ALLOWED_LIST: {},
+        constants.KEY_OF_CROSS_SILO_TIMEOUT_IN_SECONDS: 60,
+    }
+    job_config = {
+        constants.KEY_OF_GRPC_METADATA: {
+            "key": "value"
+        }
+    }
+    compatible_utils.kv.put(constants.KEY_OF_CLUSTER_CONFIG,
+                            cloudpickle.dumps(cluster_config))
+    compatible_utils.kv.put(constants.KEY_OF_JOB_CONFIG,
+                            cloudpickle.dumps(job_config))
+
+    SERVER_ADDRESS = "127.0.0.1:12344"
+    party = 'test_party'
+    cluster_parties_config = {
+        'test_party': {
+            'address': SERVER_ADDRESS,
+            'grpc_metadata': (('token', 'test-party-token'),)
```

Comment:
How does this test case make sure that this declaration actually takes effect?

Reply:
Fixed, now test recv proxy accpets expected_metadata and check it